lib/wai-bindgen-wasmer/src/le.rs

```diff
@@ -0,0 +1,195 @@
+        //   `AllBytesValid` supertrait of `Endian`.
+        unsafe {
+            assert_eq!(mem::align_of::<Le<T>>(), 1);
+            assert!(bytes.len() % mem::size_of::<Le<T>>() == 0);
+            fn all_bytes_valid<T: AllBytesValid>() {}
+            all_bytes_valid::<Le<T>>();
+
+            slice::from_raw_parts(
+                bytes.as_ptr().cast::<Le<T>>(),
+                bytes.len() / mem::size_of::<Le<T>>(),
+            )
+        }
+    }
+
+    pub(crate) fn from_slice_mut(bytes: &mut [u8]) -> &mut [Le<T>] {
+        // SAFETY: see `from_slice` above
+        //
+        // Note that both the input and the output are `mut`, helping to
+        // maintain the guarantee of uniqueness.
+        unsafe {
+            assert_eq!(mem::align_of::<Le<T>>(), 1);
+            assert!(bytes.len() % mem::size_of::<Le<T>>() == 0);
+            slice::from_raw_parts_mut(
+                bytes.as_mut_ptr().cast::<Le<T>>(),
+                bytes.len() / mem::size_of::<Le<T>>(),
+            )
+        }
+    }
+}
+
+impl<T: Copy> Clone for Le<T> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<T: Copy> Copy for Le<T> {}
+
+impl<T: Endian + PartialEq> PartialEq for Le<T> {
+    fn eq(&self, other: &Le<T>) -> bool {
+        self.get() == other.get()
+    }
+}
+
+impl<T: Endian + PartialEq> PartialEq<T> for Le<T> {
+    fn eq(&self, other: &T) -> bool {
+        self.get() == *other
+    }
+}
+
+impl<T: Endian + Eq> Eq for Le<T> {}
+
+impl<T: Endian + PartialOrd> PartialOrd for Le<T> {
+    fn partial_cmp(&self, other: &Le<T>) -> Option<Ordering> {
+        self.get().partial_cmp(&other.get())
+    }
+}
+
+impl<T: Endian + Ord> Ord for Le<T> {
+    fn cmp(&self, other: &Le<T>) -> Ordering {
+        self.get().cmp(&other.get())
+    }
+}
+
+impl<T: Endian + fmt::Debug> fmt::Debug for Le<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.get().fmt(f)
+    }
+}
+
+impl<T: Endian> From<T> for Le<T> {
+    fn from(t: T) -> Le<T> {
+        Le::new(t)
+    }
+}
+
+unsafe impl<T: AllBytesValid> AllBytesValid for Le<T> {}
+
+/// Trait used for the implementation of the `Le` type.
+pub trait Endian: AllBytesValid + Copy + Sized {
+    /// Converts this value and any aggregate fields (if any) into little-endian
+    /// byte order
+    fn into_le(self) -> Self;
+    /// Converts this value and any aggregate fields (if any) from
+    /// little-endian byte order
+    #[allow(clippy::wrong_self_convention)]
+    fn from_le(self) -> Self;
+}
+
+macro_rules! primitives {
+    ($($t:ident)*) => ($(
+        impl Endian for $t {
+            #[inline]
+            fn into_le(self) -> Self {
+                Self::from_ne_bytes(self.to_le_bytes())
+            }
+
+            #[inline]
+            fn from_le(self) -> Self {
+                Self::from_le_bytes(self.to_ne_bytes())
+            }
+        }
+    )*)
+}
+
+primitives! {
+    u8 i8
+    u16 i16
+    u32 i32
+    u64 i64
+    f32 f64
+}
```

Comment:
No u128 here?

Reply:
This was just moved over from the `wai` repo.

`u128` is not part of the component model ABI, so no point to have it here.

Reply:
Going to resolve this, feel free to comment if unclear.